src/datasets/summarizations/libs/trend.ts

```diff
@@ -141,3 +148,69 @@ export function exponentialMovingAverage(points: TimeSeriesPoint[], alpha = 0.3)
   }));
   return smoothedPoints;
 }
+
+export function centeredMovingAverage(points: TimeSeriesPoint[], k: number): TimeSeriesPoint[] {
+  const L = points.length;
+  const smoothedPoints: TimeSeriesPoint[] = [];
+  for (let i = 0; i < points.length; i++) {
+    const lSumPoints = points.slice(Math.max(0, i - k), Math.min(L, i + k));
+    const rSumPoints = points.slice(Math.max(0, i - k + 1), Math.min(L, i + k + 1));
+
+    const lSum = sum(lSumPoints.map(({ y }) => y));
+    const rSum = sum(rSumPoints.map(({ y }) => y));
+
+    const smoothedY = 0.5 * (lSum / lSumPoints.length + rSum / rSumPoints.length);
+    smoothedPoints.push({
+      x: points[i].x,
+      y: smoothedY,
+    });
+  }
+  return smoothedPoints;
+}
+
+export type GroupIdentifier = string | number;

```

Comment:
This name is very generic, could you add a comment. Is this a unique id? Why do we need a unique id?

Reply:
The explanation of parameter `groupFn`, which is the only place that uses the type `GroupIdentifier`, has been added in the JSDoc comment of `additiveDecomposite`.